test/integration/upstreams/per_host_upstream_request.cc

```diff
@@ -0,0 +1,50 @@
+#include "test/integration/upstreams/per_host_upstream_request.h"
+
+#include "common/router/router.h"
+
+using Envoy::Router::GenericConnectionPoolCallbacks;
+
+namespace Envoy {
+
+void PerHostHttpConnPool::newStream(GenericConnectionPoolCallbacks* callbacks) {
+  callbacks_ = callbacks;
+  // It's possible for a reset to happen inline within the newStream() call. In this case, we
+  // might get deleted inline as well. Only write the returned handle out if it is not nullptr to
+  // deal with this case.
```

Comment:
This comment doesn't really make sense to me since conn_pool_stream_handle_ is nullptr to begin with?

Reply:
fixed